runtime/tr.source/trj9/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -13021,6 +13023,9 @@ inlineCompareAndSwapNative(
 
    TR_X86OpCodes op;
 
+   TR::SymbolReference *callSymRef = node->getSymbolReference();
+   TR::MethodSymbol *methodSymbol = callSymRef->getSymbol()->castToMethodSymbol();
```

Comment:
I don't think these locals are necessary node->getSymbolReference()->getSymbol()->castToMethodSymbol() should be a fine chain if you need to test native-ness. Do we need to worry about a NULL return from castToMethodSymbol()? Move any locals necessary for null checking down near the use rather than declaring them up here.

Reply:
I thought there was a requirement to keep decls at the top of the block in order to keep some older compilers happy?

Reply:
True for C, but not so for the version of C++ used in the compiler. You could create a block with {}s to limit the scope if you wanted to.